### PR TITLE
Adds pagination to Blog List page

### LINKED ISSUE #140: Add pagination on the blogList page
Once we start getting a lot of blogs, we'll need to only load a few at a time & paginate the rest.

### PR DESCRIPTION
**Pull request by kendallstrautman**
Set it so each page shows 8 blogs. I first made a numbered pagination, (1,2,3...) but then after looking around at how other folks handle it I felt that select is easier to use so I switched to that. Could customize the select options more.

Fixes #140

### CODE REVIEW

File: src/templates/blog-list.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+      <Helmet>
+        <meta property="og:title" content="Blog" />
+      </Helmet>
+      <BlogList posts={data.allMarkdownRemark.edges} />
+      <FooterWrapper>
+        <Pagination>
+          <div class="prev-next">
+          {!isFirst && (
+            <Link to={prevPage} rel="prev">
+              <p>← Previous</p>
```
Comment (DirtyF) [inline src/templates/blog-list.js]:
```suggestion
              <p>← Newer</p>
```

File: src/templates/blog-list.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            </Link>
+          )}
+          {!isLast && (
+            <Link to={nextPage} rel="next">
+              <p>Next →</p>
```
Comment (DirtyF) [inline src/templates/blog-list.js]:
```suggestion
              <p>Older →</p>
```

File: src/templates/blog-list.js (hunk 1 above)
Comment (kendallstrautman) [reply src/templates/blog-list.js]: Nice, that is better 👌🏼

Comment (DirtyF) [reply src/templates/blog-list.js]: You often see `Newer`/ `Older` (posts), like on https://www.gatsbyjs.org/blog/

File: src/templates/blog-list.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+              <p>Older →</p>
+            </Link>
+          )}
+          </div>
+          <div class="list-numbers">
+            <ul>
+              <PaginationSelect>
+                <p>Page</p>
+                  <div class="select">
+                    <select value={selectValue} onChange={handleSelectChange}>
```
Comment (DirtyF) [inline src/templates/blog-list.js]:
```suggestion
                    <select aria-label="Pagination Dropdown" value={selectValue} onChange={handleSelectChange}>
```

File: src/templates/blog-list.js [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+                      {Array.from({length: numPages}, (_, i) => (
+                        <option value={i + 1}>
```
Comment (DirtyF) [inline src/templates/blog-list.js]:
```suggestion
                        <option arial-label="Goto Page {i + 1}" value={i + 1}>
```

File: src/templates/blog-list.js [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                          {i + 1}
+                        </option>
```
Comment (DirtyF) [inline src/templates/blog-list.js]: Would be even better if we can convey if we're on the current page or not with [aria-current](https://www.w3.org/TR/wai-aria-1.1/#aria-current) set to `true `or `false`

Comment (kendallstrautman) [reply src/templates/blog-list.js]: good call, I can add that

**Comment by DirtyF** [comment]
Thanks for taking care of a11ty 🙏 

Small rendering glitch on Safari:
page-dropdown-safari

**Comment by kendallstrautman** [comment]
Good catch, should be fixed now!

**Comment by DirtyF** [comment]
Nice 😄 

I just noticed the render of the focus state looks a bit _shrinky_ 😊 
Capture d’écran 2019-12-21 à 10 42 11

compared to
Capture d’écran 2019-12-21 à 10 42 41

**Pull request merged**